### DISCUSSION TITLE
Error Running Migrated PDI to Hop Project: Works Locally but Fails Remotely

### DISCUSSION THREAD

**Discussion by ravamo** [apache/hop#4685] [Q&A] [answered]
I checked a project migrated from PDI to Hop using the tool, and everything seems fine. However, when I execute it as mentioned in the [thread ](https://github.com/apache/hop/discussions/4634), I get this error, and I don’t know why. I searched for this error on Google and found this [thread ](https://www.mail-archive.com/users@hop.apache.org/msg00079.html), but it's not clear to me why it happens, especially since it runs perfectly fine locally.

```
Starting action
Opening pipeline: [/servers/hop/server/resources/etl/repository/setExecutionTime.hpl]
Starting pipeline...(file=/servers/hop/server/resources/etl/repository/setExecutionTime.hpl, name=Set Execution time, repinfo=null)
ERROR: Unable to open pipeline:
This action needs a run configuration to use to execute the specified pipeline
ERROR: org.apache.hop.core.exception.HopException:
This action needs a run configuration to use to execute the specified pipeline
        at org.apache.hop.workflow.actions.pipeline.ActionPipeline.execute(ActionPipeline.java:491)
        at org.apache.hop.workflow.Workflow.executeFromStart(Workflow.java:745)
        at org.apache.hop.workflow.Workflow.executeFromStart(Workflow.java:888)
        at org.apache.hop.workflow.Workflow.executeFromStart(Workflow.java:447)
        at org.apache.hop.workflow.Workflow.startExecution(Workflow.java:307)
        at org.apache.hop.workflow.engines.local.LocalWorkflowEngine.startExecution(LocalWorkflowEngine.java:248)
        at java.base/java.lang.Thread.run(Thread.java:833)
 mainRedmineLoadDatalake - Starting action [Error Handling]
 mainRedmineLoadDatalake - exec(5, 0, Error Handling)
Starting action
Opening pipeline: [/servers/hop/server/resources/etl/repository/errorHandling.hpl]
Starting pipeline...(file=/servers/hop/server/resources/etl/repository/errorHandling.hpl, name=Error Handling, repinfo=null)
ERROR: Unable to open pipeline:
This action needs a run configuration to use to execute the specified pipeline
ERROR: org.apache.hop.core.exception.HopException:
This action needs a run configuration to use to execute the specified pipeline
         at org.apache.hop.workflow.actions.pipeline.ActionPipeline.execute(ActionPipeline.java:491)
         at org.apache.hop.workflow.Workflow.executeFromStart(Workflow.java:745)
        at org.apache.hop.workflow.Workflow.executeFromStart(Workflow.java:888)
         at org.apache.hop.workflow.Workflow.executeFromStart(Workflow.java:447)
         at org.apache.hop.workflow.Workflow.startExecution(Workflow.java:307)
       at org.apache.hop.workflow.engines.local.LocalWorkflowEngine.startExecution(LocalWorkflowEngine.java:248)
         at java.base/java.lang.Thread.run(Thread.java:833)
 mainRedmineLoadDatalake - Starting action [Abort job]
````

**Comment by bamaer** [comment]
It looks like your `mainRedmineLoadDatalake` workflow has a pipeline action for `/servers/hop/server/resources/etl/repository/errorHandling.hpl` that doesn't have a run configuration or one that the server doesn't recognize.

**Reply by ravamo** [reply]
Hi ! 
Actually, there are two ETLs that fail to execute, but I don't understand why. They work perfectly fine on the local machine, but on the server, they throw this error and stop. I've checked to ensure that all the metadata is in the folder, and it is. I don't quite understand where the problem lies. Is there any additional step required for migrating ETLs from PDI to Hop that I might have missed? (I used the migration tool, added the database configuration, and set the parameters).

**Reply by NicoDupont** [reply]
did you set the run configuration ?
![image](https://github.com/user-attachments/assets/2925fdbb-a989-4b02-8706-5bb0beec0760)

**Reply by ravamo** [reply]
yep
![image](https://github.com/user-attachments/assets/bb9a507e-6e8a-46cb-a79d-fea39c87cb49)

**Reply by NicoDupont** [reply]
try with the remote run configuration not local

**Reply by bamaer** [reply]
and make sure you have both a `local` workflow run config AND a `local` pipeline run config on the server.

**Reply by NicoDupont** [reply]
![image](https://github.com/user-attachments/assets/d5a97287-7072-4dca-8ef9-7c96aa79a792)

**Reply by ravamo** [reply]
I've this configuration :
![image](https://github.com/user-attachments/assets/c451c4eb-b2fd-4a37-a08c-856cb14d378f)
And I’ve checked, and all of them have the "local" mode set both on my local machine and on the server, for both pipelines and workflows.

**Reply by bamaer** [reply]
did you check the XML for the problematic workflow? we had an issue with Hop Gui showing the default run config, even if it wasn't serialized. Selecting the run config for that action and saving the workflow fixed that issue, and that issue should have been resolved ages ago, but you never know.

**Reply by ravamo** [reply]
Well, look at that! I ran a grep to search for the word "local" within the XML files, and there’s nothing. I understand that’s not right, correct? I mean, each XML should have a tag indicating that, right?

**Reply by bamaer** [reply]
yes, just create a backup of that failing workflow, open it in Hop Gui, select the run config for that pipeline action, save your workflow and compare that last version's XML to the XML from the backup and you'll know if that is the issue

**Reply by ravamo** [reply]
Well, I deleted everything and recreated it manually, and it seems to be working now—or at least I’m getting new errors because I set everything to debug mode. Now I’m dealing with this error:
``` less
ERROR: Error initializing transform [Salida Tabla]
ERROR: java.lang.NullPointerException: Cannot invoke "org.apache.hop.core.database.DatabaseMeta.getIDatabase()" because "this.data.databaseMeta" is null
         at org.apache.hop.pipeline.transforms.tableoutput.TableOutput.init(TableOutput.java:482)
         at org.apache.hop.pipeline.transform.TransformInitThread.run(TransformInitThread.java:66)
         at java.base/java.lang.Thread.run(Thread.java:833)
repository\setExecutionTime - Transform [Información de sistema.0] initialized flawlessly.
```` 
But the datasoure was defined .. i'm lost right now

**Reply by bamaer** [reply]
it looks like that database connection isn't known on the server though

**Reply by ravamo** [reply, marked as answer]
yep looks list a problem with rdbs ,  but I added the JSON that I created locally (it’s parameterized) to the folder hop/config/projects/default/metadata/rdbms, and then I added the path in the XML file before starting the server:
```xml
<metadata_folder></metadata_folder>
```

**Reply by bamaer** [reply]
not something I use frequently myself, but I would avoid the metadata folders for the `samples` and `default` projects for your own metadata. anyway, this is now a configuration issue and no longer a PDI migration issue.

**Reply by ravamo** [reply]
Yes, but now I’m lost... Honestly, I’m not sure what tests I can run. I think it’s better if I close this one and open a new one for this issue, right?

**Reply by bamaer** [reply]
yes, that's probably better to avoid having two separate topics in a single discussion
